lib/client/api_test.go

```diff
@@ -1214,3 +1216,32 @@ func TestConnectToProxyCancelledContext(t *testing.T) {
 	require.Nil(t, clusterClient)
 	require.Error(t, err)
 }
+
+func TestIsErrorResolvableWithRelogin(t *testing.T) {
+	for _, tt := range []struct {
+		name             string
+		err              error
+		expectResolvable bool
+	}{
+		{
+			name:             "private key policy error should be resolvable",
+			err:              keys.NewPrivateKeyPolicyError(keys.PrivateKeyPolicyHardwareKey),
+			expectResolvable: true,
+		}, {
+			name: "wrapped private key policy error should be resolvable",
+			err: &interceptors.RemoteError{
+				Err: keys.NewPrivateKeyPolicyError(keys.PrivateKeyPolicyHardwareKey),
+			},
+			expectResolvable: true,
+		},
+	} {
+		t.Run(tt.name, func(t *testing.T) {
+			resolvable := IsErrorResolvableWithRelogin(tt.err)
+			if tt.expectResolvable {
+				require.True(t, resolvable, "Expected error to be resolvable with relogin")
+			} else {
+				require.False(t, resolvable, "Expected error to be unresolvable with relogin")
+			}
```

Comment:
Maybe? The error message is a bit worse, though.

```suggestion
			resolvable := IsErrorResolvableWithRelogin(tt.err)
      assert.Equal(t, tt.expectResolvable, resolvable, "IsErrorResolvableWithRelogin() mismatch")
```

Reply:
Yeah that's what I started with, but I'd rather maintain the clearer error message.